### PR TITLE
feat: CREATE and CREATE2 support [APE-1086]

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

Support CREATE all over

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: evm_trace/display.py [hunk 1]
```diff
@@ -46,7 +46,12 @@ def title(self) -> str:
         call_path = str(address) if address else ""
         if self.call.calldata:
             call_path = f"{call_path}." if call_path else ""
-            call_path = f"{call_path}<{self.call.calldata[:4].hex()}>"
+            method_id = (
+                "__init__"
```
Comment (antazoey) [inline evm_trace/display.py:50]: ethtx.info does `New()` here but I kind of like `__init__`

Comment (antazoey) [reply evm_trace/display.py:50]: actuallly.... realizing `__new__` makes more sense

**Comment by antazoey** [comment]
ISSUE: The parity style constructors are missing inputs. I can see the data on he field called `init` but so is the bytecode

**Comment by antazoey** [comment]
- [x] geth Calltracer support
- [x] parity support
- [ ] geth structlog spport

**Review by fubuloubu — APPROVED** [review]

**Comment by fubuloubu** [comment]
Might need some additional tests for CREATE2 support

**Comment by antazoey** [comment]
create2 aint fully working yet but should be soon, and ill include tests with it, putting back in draft

**Comment by fubuloubu** [comment]
> create2 aint fully working yet but should be soon, and ill include tests with it, putting back in draft

Okay, if that's the case just link an issue for adding it

**Comment by antazoey** [comment]
> Okay, if that's the case just link an issue for adding it

I think I can at least get some minimal support for CREATE2 and i will create an issue for the remainder if there is any.
Also, needing to fix CREATE (and CREATE2) happening inside the contract during structLog style trace -> call Tree creation

**Comment by antazoey** [comment]
@fubuloubu OK I think this is finally ready, I tested on every provider.

File: evm_trace/geth.py [hunk 2]
```diff
@@ -198,10 +199,17 @@ def _create_node(
             node_kwargs.get("last_create_depth")
             and frame.depth == node_kwargs["last_create_depth"][-1]
         ):
+            # If we get here, we are in the process of completing the attributes from
+            # a CREATE(2) node. The data is located at the first frame with the same depth
+            # after the CREATE(2) opcode was found. This idea is copied from Brownie.
```
Comment (fubuloubu) [inline evm_trace/geth.py:207]:
```suggestion
            # a `CREATE2` node. The data is located at the first frame with the same depth
            # after the `CREATE2` opcode was found. This idea is copied from Brownie.
```

also maybe link the code?

**Comment by fubuloubu** [comment]
Other than that one change, it looks good to merge

File: evm_trace/geth.py (hunk 2 above)
Comment (antazoey) [reply evm_trace/geth.py:207]: I was trying to say that this is for both CREATE and CREATE2 but i guess that was not clear

Comment (fubuloubu) [reply evm_trace/geth.py:207]: the opcode mnemonic is just `CREATE2`
https://eips.ethereum.org/EIPS/eip-1014

Comment (antazoey) [reply evm_trace/geth.py]: i clarifizzled

**Pull request merged**